diff -rupN original/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml patch/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
--- original/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml	2017-03-01 07:49:54.000000000 +0000
+++ patch/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml	2017-03-16 07:33:42.000000000 +0000
@@ -260,6 +260,46 @@ SilicaFlickable {
             }
         }
 
+        ConfigurationGroup {
+            id: lockscreenUpcomingSettings
+            path: "/desktop/lipstick-jolla-home/lockscreenUpcoming"
+            property bool fixClock: true
+            property int xPos: 0
+            property int yPos: 0
+        }
+        ShowCalEv
+        {
+           id: showcalevents
+           anchors 
+           {
+               top: lockscreenUpcomingSettings.fixClock ?  clock.bottom : undefined
+               topMargin:lockscreenUpcomingSettings.fixClock ?  Theme.paddingMedium: undefined
+               right: lockscreenUpcomingSettings.fixClock ? parent.right: undefined
+               rightMargin: lockscreenUpcomingSettings.fixClock ?  Theme.paddingMedium : undefined
+            }
+           x: lockscreenUpcomingSettings.fixClock ? undefined: lockscreenUpcomingSettings.xPos 
+          y : lockscreenUpcomingSettings.fixClock ? undefined: lockscreenUpcomingSettings.yPos         
+           state: "hidden"
+           states: [
+               State 
+               {
+                   name: "hinting"
+                   when: lockScreen.locked && !lockScreen.panning && lockItem.allowAnimations
+                   StateChangeScript
+                   {
+                       //name: "myScript"
+                       script:
+                       {
+                           showcalevents.runAnimations()
+                       }
+                   }                   
+               },
+               State
+               {
+                   name: "hidden"
+               }
+           ]
+        }
         Clock {
             id: clock
 
